### ISSUE TITLE
Ollama not working in 2.1.0

### ISSUE THREAD

**Issue by Zyfax** [longy2k/obsidian-bmo-chatbot#83]
After upgrading to the latest, I encountered an issue where Ollama API stopped functioning properly.

I made sure that enable_reference_current_note and enable_header were updated. To troubleshoot further, I even removed and reinstalled the plugin.

However, when I attempted to communicate with BMO through chat, it responded with a "Connection not found" error message. Meanwhile, the shortcut `CTRL+SHIFT+=` yielded a "Generation complete" message, suggesting that some processing was occurring.

In addition, under Settings > General > Model, the Models dropdown list is currently empty. Furthermore, clicking the Reload button results in the message "Models reloaded," but no changes appear to have been made.

**Comment by longy2k** [comment]
Check out: [How to setup with Ollama · longy2k/obsidian-bmo-chatbot Wiki](https://github.com/longy2k/obsidian-bmo-chatbot/wiki/How-to-setup-with-Ollama)

I have updated Ollama to use the javascript API which now requires additional setup to bypass the CORS policy in Obsidian.

Please let me know if you have resolved this.

**Comment by Zyfax** [comment]
Yep, first thing I did.
My Ollama is running on a different computer under docker but got the env: OLLAMA_ORIGINS="*".
I can also see that Ollama is receiving 200 request each time.

**Comment by longy2k** [comment]
I am assuming you are now able to successfully run Ollama with BMO Chatbot?

Please close this issue if you got it working.

Thanks! :D

**Comment by Zyfax** [comment]
No it is still not working.
BMO can't get the models in the dropdown in settings and chat still gives me "Connection not found."

**Comment by longy2k** [comment]
If you are on MacOS, try running the first option without the docker to see if it is working.

Make sure you quit the Ollama application completely and only run the `OLLAMA_ORIGINS=app://obsidian.md* ollama serve` command in the terminal. Then re-insert `http://localhost:11434` under the OLLAMA REST API URL.

Otherwise, check out: https://medium.com/dcoderai/how-to-handle-cors-settings-in-ollama-a-comprehensive-guide-ee2a5a1beef0

**Comment by Zyfax** [comment]
Im running Ollama on a linux server with docker.  This is the environments defined, `OLLAMA_ORIGINS="*"` and `OLLAMA_HOST="0.0.0.0:11434"`. Even testet with `OLLAMA_ORIGINS=app://obsidian.md*`.

**Comment by aAbdman** [comment]
same problem here

**Comment by longy2k** [comment]
Are you running into any port issues (e.g. `address already in use` or `port already in use`)?

I just realized that handling ports might be a common issue and is a bit of a hassle.

**Comment by Zyfax** [comment]
No, Ollama is working fine with Open WebUI, curl and Python form remote sources.

Can confim that BMO is working with a local hosted Ollama, but not remote(same LAN) as my setup.
Server is running RTX 3070Ti something that my laptop is not.

**Comment by twalderman** [comment]
I tested it with a fresh vault. It is not working correctly.  I can see the server response to the addition of the config. No model loading. I tried the old /v1 - no love.


[GIN] 2024/05/23 - 19:59:06 | 200 |     493.875µs |       127.0.0.1 | GET      "/"
[GIN] 2024/05/23 - 19:59:15 | 200 |      34.583µs |       127.0.0.1 | GET      "/"
[GIN] 2024/05/23 - 19:59:21 | 200 |      28.417µs |       127.0.0.1 | GET      "/"
[GIN] 2024/05/23 - 19:59:28 | 200 |      14.792µs |       127.0.0.1 | GET      "/"
[GIN] 2024/05/23 - 19:59:28 | 404 |      16.334µs |       127.0.0.1 | GET      "/v"
[GIN] 2024/05/23 - 19:59:29 | 404 |       6.792µs |       127.0.0.1 | GET      "/v1"

**Comment by twalderman** [comment]
Im on a Mac.  I got it to work on vault. not exactly sure why because it doesnt work on fresh vault.  It only works when the server is 11434. Any other port will not retrieve models. The ollama app is not running to avoid port conflict. OLLAMA_ORIGINS=app://obsidian.md* OLLAMA_HOST=127.0.0.1:11434

**Comment by Zyfax** [comment]
I have been testing different approaches.

It looks like when you refresh models, it sends two requests. One goes to http://127.0.0.1:11434 and the other to the Ollama REST API URL defined in the settings.
If you are running a local Ollama instance, it will receive the models and then work, but it will use the local Ollama connection.

If I'm running an SSH tunnel with `ssh user@remotehost -L 11434:127.0.0.1:11434`, it works, but it is not possible to change the port as @twalderman mentioned.

Inside the plugin in the Vault at .obsidian\plugins\bmo-chatbot\main.js, there is a line at 1644 with the default host. If I change that to my remote server and restart Obsidian, it then works.
1644: `this.config.host = formatHost((_a2 = config == null ? void 0 : config.host) != null ? _a2 : "http://remotehost:11434");`

It is possible that BMO is ignoring the Ollama REST API URL and falling back to the default value (http://127.0.0.1:11434).
